### PR TITLE
Add iframe editor header toolbar and block inserter

### LINKED ISSUE #38486: Product Block Editor: Add modal editor toolbar and block appender
This issue will involve creating a fixed toolbar with a block appender button above the editor contextual toolbar inside the modal editor.  Note that this logic may live inside the modal editor or may be better suited inside the `IframeEditor` component, but styling will need to be added for the modal editor regardless.

### Design

AkVNGImLgSqCObTQ3idVn7-fi-305_38548

Image

### Acceptance criteria

* [ ] A fixed toolbar is shown above the contextual toolbar in the modal editor
* [ ] The block appender button is shown in this toolbar, regardless of block selection
* [ ] The block appender should allow adding of any of the enabled iframe editor blocks

### PR DESCRIPTION
**Pull request by joshuatf**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

* Adds a secondary sidebar to the iframe editor
* Adds a header toolbar
* Adds the block inseter/appender

Screen Shot 2023-05-31 at 9 32 00 AM

Closes #38486 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product block editor under WooCommerce -> Settings -> Advanced -> Features
2. Navigate to the "Add product" page
3. Click on "Add description" in the General tab
4. Note the new header with the block inserter (blue "+" button)
5. Add some blocks, verifying they get inserted correctly

### CODE REVIEW

File: packages/js/product-editor/src/components/iframe-editor/iframe-editor.tsx [hunk 1]
```diff
@@ -45,6 +47,7 @@ export function IframeEditor( {
 }: IframeEditorProps ) {
 	const [ resizeObserver, sizes ] = useResizeObserver();
 	const [ blocks, setBlocks ] = useState< BlockInstance[] >( initialBlocks );
+	const [ isInserterOpened, setIsInserterOpened ] = useState( false );
```
Comment (joshuatf) [inline packages/js/product-editor/src/components/iframe-editor/iframe-editor.tsx:50]: Noting that in Gutenberg this is retrieved from the `editPostStore` store or `editWidgetsStore`, both of which feel wrong for this use case, though we may consider moving this into the editor settings or context at some point if necessary.

File: packages/js/product-editor/src/components/modal-editor/style.scss [hunk 2]
```diff
@@ -10,6 +10,8 @@ $modal-editor-height: 60px;
     .components-modal__header {
         height: $modal-editor-height;
         border-bottom: 1px solid $gray-400;
+        padding-left: 18px;
+        padding-right: 18px;
```
Comment (louwie17) [inline packages/js/product-editor/src/components/modal-editor/style.scss:14]: Is there a reason we are not following the divisible by 4 space here? 4, 8, 16, 32?

**Review by louwie17 — APPROVED** [review]
Nice work!! This overall tested well, I did notice one issue where the inner scroll doesn't work for the sidebar block library, which is kinda weird (can be fixed as a follow up).
I did leave one additional question in the code.

For the rest I assume quite a bit of this is copy/paste from Gutenberg?

File: packages/js/product-editor/src/components/modal-editor/style.scss (hunk 2 above)
Comment (joshuatf) [reply packages/js/product-editor/src/components/modal-editor/style.scss:14]: Good question. This is just what I noticed in the design.  Do we want to change this, @jarekmorawski?

Comment (joshuatf) [reply packages/js/product-editor/src/components/modal-editor/style.scss:14]: Going to get this merged to unblock other work.  @jarekmorawski if you want to modify this, please let me know and we can create an issue to adjust.

**Pull request merged**